app/Models/Category.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use App\Models\Product;
+use App\Models\Suggest;
+
+class Category extends Model
+{
+    protected $table = 'categories';
+    protected $fillable = [
+        'category_name',
+    ];
+
+    public function product()
+    {
+        return $this->hasMany(Product::class, 'product_id', 'id');
+    }
+
+    public function suggest()
+    {
+        return $this->hasMany(Suggest::class, 'suggest_id', 'id');
```

Comment:
Chỗ này a không dùng tham số thứ 2 và 3 ('suggest_id', 'id') có được không?
Nếu a chỉ dùng `return $this->hasMany(Suggest::class); `thì sẽ có ý nghĩa như nào?

Reply:
Dạ, theo em nghĩ không có dùng tham số 2, và tham số 3 ('suggest_id', 'id') vẫn được.
Nếu dùng return $this->hasMany(Suggest::class);  thì foreign key sẽ tự biết được mối liên kết giữa các bảng theo naming convention.

Reply:
Trả lời đúng. Lararvel tự nhận trường [model]_id làm khóa ngoại khi khai báo Relationship, e đang đặt tên đúng chuẩn rồi nên k cần tham số thứ 2,3 nó cũng sẽ hiểu khi join.
Tuy nhiên chỗ này e viết sai. 
return $this->hasMany(Suggest::class, 'suggest_id', 'id');
Thế này mới đúng: 
return $this->hasMany(Suggest::class, 'category_id', 'id');
category_id là khóa ngoại ở bảng Subject để xác định Subject đó thuộc Category nào. Và Category này có những Subject nào. Chứ k có cái nào là subject_id ở bảnh Category và Subject cả.

A check 1 lượt belongsTo thì e khai báo foregin đúng. Còn hasMany có 1 số cái nhầm lẫn e self-check lại he.